github/examples_test.go

```diff
@@ -77,7 +78,11 @@ func ExampleUsersService_ListAll() {
 
 func ExamplePullRequestsService_Create() {
 	// In this example we're creating a PR and displaying the HTML url at the end.
-	client := github.NewClient(nil)
+	ctx := context.Background()
+	token := "MyGitHubTokenThatIDontStoreInMyScripts"
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
```

Comment:
Here's an idea, in addition to adding the comment saying authentication is needed and linking to its documentation, you can keep this auth code, just factor it out into a helper:

```Go
client := newGitHubClientWithAuth()
```

And below `ExamplePullRequestsService_Create`, you can create that helper:

```
func newGitHubClientWithAuth() *github.Client {
	ts := oauth2.StaticTokenSource(
		&oauth2.Token{AccessToken: "... your access token ..."},
	)
	tc := oauth2.NewClient(context.Background(), ts)
	return github.NewClient(tc)
}
```

That way the code is realistic, and the details of authentication don't need to detract from the main point of the `ExamplePullRequestsService_Create` example.

Reply:
But you can't see the code of `newGitHubClientWithAuth` in the final godoc if you do that right?
So that would be a bit misleading for the user that thinks the `newGitHubClientWithAuth` is an unexported helper from the package right?

Reply:
You're right, it shouldn't show up. To see the actual source code of `newGitHubClientWithAuth`, it would require people to jump into the source code.